### PR TITLE
Point at the right enclosing scope when using `await` in non-async fn

### LINKED ISSUE #63398: Wrong block referenced when emitting E0728
The following code ([Playground link](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=3e36922f1d43ba54d23783769a15e381))

```rust
#![feature(async_await)]
async fn do_the_thing() -> u8 {
    8
}

fn main() {
    let x = move || {};
    let y = do_the_thing().await;
}
```

Emits the incorrect error:
```
   Compiling playground v0.0.1 (/playground)
error[E0728]: `await` is only allowed inside `async` functions and blocks
 --> src/main.rs:8:13
  |
7 |     let x = move || {};
  |             ------- this is not `async`
8 |     let y = do_the_thing().await;
  |             ^^^^^^^^^^^^^^^^^^^^ only allowed inside `async` functions and blocks

error: aborting due to previous error

error: Could not compile `playground`.

To learn more, run the command again with --verbose.
```

It should point to the block belonging to `fn main` and not the closure from the line above the usage of `await`.

### PR DESCRIPTION
**Pull request by estebank**
Fix #63398.

### CODE REVIEW

**Comment by pietroalbini** [comment]
This is beta-nominated for 1.38 right?

**Comment by estebank** [comment]
@pietroalbini correct, I'm aware of the current release and it'd be against procedure to include it in 1.37.

**Comment by pietroalbini** [comment]
Great! I was a bit confused since 1.38 didn't branch out to beta yet :sweat_smile:

File: src/test/ui/issues/issue-63398.rs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+// edition:2018
```
Comment (Centril) [inline src/test/ui/issues/issue-63398.rs]: Move this to `src/test/ui/async-await/issues/`, include a short description of the problem in the filename, and also please add a comment to the test.

**Comment by Centril** [comment]
r=me if you like with nit ^-- addressed

**Comment by Centril** [comment]
@estebank  could you try to see if `--bless --compare-mode=nll` makes any `git diff`? https://github.com/rust-lang/rust/pull/63563#issuecomment-521365857 may have failed due to this PR but I'm not sure.

**Comment by estebank** [comment]
Can't see how it could have, but I'll do so soon.

**Comment by Centril** [comment]
Yeah no; this PR is not at fault.

**Comment by nikomatsakis** [comment]
We decided not to accept this for beta since async-await didn't make the beta cu.

**Pull request merged**